dali/kernels/common/transpose.h

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_COMMON_TRANSPOSE_H_
+#define DALI_KERNELS_COMMON_TRANSPOSE_H_
+
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#include "dali/core/static_switch.h"
+#include "dali/core/tensor_view.h"
+#include "dali/kernels/common/utils.h"
+
+namespace dali {
+namespace kernels {
+
+namespace detail {
```

Comment:
`detail` namespace does not isolate a particular functionality from the entire codebase - as such, potentially ambiguous names should be avoided in it (vide CollapseShape) - I'd recommend renaming it to sth like `transpose_impl`.

Reply:
Done